### PR TITLE
Add language: Sweave 

### LINKED ISSUE #6538: New Language: Sweave
## Language name

Sweave 

A `prose` language for literate programming with the R statistical software language. Precursor to Rmarkdown, a supported language. Mixes R code with TeX documentation. Present in documentation of many R language repositories.

## URL of example repository

17.9k files
https://github.com/search?q=path%3A*.rnw+NOT+randomstring154769&type=code

https://github.com/rstudio/rstudio
https://github.com/lme4/lme4

## URL of syntax highlighting grammar

https://github.com/textmate/sweave.tmbundle

## Most popular extensions

- .Rnw
- .Snw
- .rnw
- .snw

## Detected language

none

### PR DESCRIPTION
**Pull request by grady**
## Description

resolves #6538

A prose language for literate programming with the R statistical software language. Precursor to Rmarkdown, a supported language. Mixes R code with TeX documentation. Present in documentation of many R language repositories.

## Checklist:

- [x] **I am adding a new language.**
  - [x] The extension of the new language is used in hundreds of repositories on GitHub.com.
    - Search results for each extension:
      - [.Rnw](https://github.com/search?type=code&q=NOT+is%3Afork+path%3A*.rnw)
  - [x] I have included a real-world usage sample for all extensions added in this PR:
    - Sample source(s):
      - [knitr-beamer.Rnw](samples/Sweave/knitr-beamer.Rnw) source: https://github.com/yihui/knitr/blob/master/inst/examples/knitr-beamer.Rnw
      - [knitr-minimal.Rnw](samples/Sweave/knitr-minimal.Rnw) source: https://github.com/yihui/knitr/blob/master/inst/examples/knitr-minimal.Rnw
    - Sample license(s): GPL for all
  - [x] I have included a syntax highlighting grammar: https://github.com/textmate/sweave.tmbundle
      
  - [x] I have added a color
    - Hex value: `#198ce7`
    - Rationale: Same color as R and Rmarkdown, since this is a related language
  - [ ] I have updated the heuristics to distinguish my language from others using the same extension.
    - these extensions are unused by other languages at this time

### CODE REVIEW

**Comment by grady** [comment]
Questionable if the `.Snw` extension should be included or not. It is obsolete and only has 350 hits in the search, while the dominant `.Rnw` extension has over 17k results.

**Comment by lildude** [comment]
> Questionable if the `.Snw` extension should be included or not. It is obsolete and only has 350 hits in the search, while the dominant `.Rnw` extension has over 17k results.

Not 😄 It doesn't meet our popularity requirements by a large margin. An override can always be used once this PR have been merged and deployed.

**Review by lildude — CHANGES_REQUESTED** [review]
Please remove or replace `samples/Sweave/Sweave.Rnw` with a smaller sample as the current sample is unnecessarily large (if it's not doodles in the diff by default, it's generally too big).

File: vendor/licenses/git_submodule/sweave.tmbundle.dep.yml [hunk 1]
```diff
@@ -0,0 +1,17 @@
+---
+name: sweave.tmbundle
+version: d69ff113e77af07d46624095c413f5e60572a8fe
+type: git_submodule
+homepage: https://github.com/textmate/sweave.tmbundle.git
+license: other
```
Comment (lildude) [inline vendor/licenses/git_submodule/sweave.tmbundle.dep.yml:6]:
```suggestion
license: permissive
```

**Comment by grady** [comment]
Removed .snw extension from languages.yml and deleted related .snw sample file.
Removed large Sweave.Rnw sample file. Two samples remain.
Edited PR description to match above.

As far as heuristics go, the unique features of the language that I can think of are
- basic structure of a tex file, in fact it could be a valid tex file itself with nothing extra.
- the string `\Sexpr` anywhere in the file (optional, not extremely common)
- lines that begin with `<<` and end with `>>=` (also optional, but most files will have this)

**Comment by lildude** [comment]
Don't worry about the heuristics… s as you've pointed out this is the only user of that extension and a quick search suggests this isn't likely to cause problems.

You will however need to address the test failures.

**Comment by grady** [comment]
I addressed the ace_mode fail, but the request to run a script will need to wait until tomorrow. Dev machine is in the office.

I set ace_mode to tex, on the basis that a valid .tex file is also valid .Rnw.

**Pull request merged**